Issue with json struct tag omitempty

I found this bug in gocsv: https://github.com/gocarina/gocsv/issues/263
and thought I would test the same on your project, since it seems better in many ways.
But unfortunately it seems that this project has the same bug, or at least what I think is a bug.

The error is that the code panics when having json struct tags with omitempty on them. I haven't investigated very deeply, but it seems like it should not panic.

Example panic output:
```
--- FAIL: TestCSV (0.00s)
    --- FAIL: TestCSV/test_1_json (0.00s)
panic: reflect: call of reflect.Value.Field on zero Value [recovered]
	panic: reflect: call of reflect.Value.Field on zero Value

```

Test code I used:
```golang
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"testing"

	"github.com/shigetaichi/xsv"
)

type TestObj struct {
	Name    string     `json:"name,omitempty"`
	SubObj  *TestOb2   `json:"subObj,omitempty"`
	SubObjs []*TestOb2 `json:"subObjs,omitempty"`
}

type TestOb2 struct {
	A string `json:"a,omitempty"`
	B string `json:"b,omitempty"`
}

func TestCSV(t *testing.T) {
	obj1 := []*TestObj{
		{
			Name: "Obj1",
			SubObj: &TestOb2{
				A: "a1",
				B: "b1",
			},
			SubObjs: []*TestOb2{
				{
					A: "a2",
					B: "b2",
				},
				{
					A: "a3",
					B: "b3",
				},
			},
		},
	}

	testCases := []struct {
		TagName string
	}{
		{"json"}, {"csv"},
	}
	for i, tc := range testCases {
		tc := tc
		t.Run(fmt.Sprintf("test_%d_%s", i+1, tc.TagName), func(t *testing.T) {
			xsvWrite := xsv.NewXsvWrite[*TestObj]()
			xsvWrite.TagName = tc.TagName
			buf := bytes.NewBuffer(nil)
			err := xsvWrite.SetBufferWriter(buf).Write(obj1)
			if err != nil {
				t.Error(err)
			}
			fmt.Println("Buf:", buf.String())
			obj2 := []*TestObj{}
			xsvReader := xsv.NewXsvRead[*TestObj]()
			xsvReader.TagName = tc.TagName
			err = xsvReader.SetStringReader(buf.String()).ReadTo(&obj2)
			if err != nil {
				t.Error(err)
			}
			if fmt.Sprint(obj1) != fmt.Sprint(obj2) {
				t.Errorf("objects not equal. Obj1: %v, Obj2: %v", obj1, obj2)
			}
		})
	}
}

func (o *TestObj) String() string {
	data, err := json.Marshal(o)
	if err != nil {
		panic(err)
	}
	return string(data)
}

```

--- a/decode_test.go
+++ b/decode_test.go
@@ -223,6 +223,26 @@ ff,gg,22,hh,ii,jj`)
 	}
 }
 
+func Test_readTo_embed_ptr_all_omitempty(t *testing.T) {
+	b := bytes.NewBufferString(`first,last,abc,Omit
+aa,dd,ee,`)
+
+	var rows []EmbedPtrAllOmitemptySample
+	if err := NewXsvRead[EmbedPtrAllOmitemptySample]().SetReader(csv.NewReader(b)).ReadTo(&rows); err != nil {
+		t.Fatalf(err.Error())
+	}
+	expected := EmbedPtrAllOmitemptySample{
+		Qux: "aa",
+		AllOmitEmpty: &AllOmitEmpty{
+			Omit: nil,
+		},
+		Quux: "dd",
+	}
+	if !reflect.DeepEqual(expected, rows[0]) {
+		t.Fatalf("expected first sample %v, got %+v", expected, rows[0])
+	}
+}
+
 func Test_readTo_slice(t *testing.T) {
 	b := bytes.NewBufferString(`Slice
 []

--- a/sample_structs_test.go
+++ b/sample_structs_test.go
@@ -15,6 +15,12 @@ type Sample struct {
 	Omit *string `csv:"Omit,omitempty"`
 }
 
+type AllOmitEmpty struct {
+	Omit  *string `csv:"Omit,omitempty"`
+	Omit2 *string `csv:"Omit2,omitempty"`
+	Omit3 *string `csv:"Omit3,omitempty"`
+}
+
 type SliceSample struct {
 	Slice []int `csv:"Slice"`
 }
@@ -96,6 +102,14 @@ type EmbedUnmarshalCSVWithClashingField struct {
 type EmbedPtrSample struct {
 	Qux string `csv:"first"`
 	*Sample
+	Ignore string  `csv:"-"`
+	Grault float64 `csv:"garply"`
+	Quux   string  `csv:"last"`
+}
+
+type EmbedPtrAllOmitemptySample struct {
+	Qux string `csv:"first"`
+	*AllOmitEmpty
 	Ignore string  `csv:"-"`
 	Grault float64 `csv:"garply"`
 	Quux   string  `csv:"last"`

--- a/types.go
+++ b/types.go
@@ -217,10 +217,12 @@ func toFloat(in interface{}) (float64, error) {
 
 func setField(field reflect.Value, value string, omitEmpty bool) error {
 	if field.Kind() == reflect.Ptr {
-		if omitEmpty && value == "" {
-			return nil
-		}
 		if field.IsNil() {
+			if omitEmpty && value == "" {
+				if field.Type().Elem().Kind() != reflect.Struct {
+					return nil
+				}
+			}
 			field.Set(reflect.New(field.Type().Elem()))
 		}
 		field = field.Elem()

